src/Microsoft.DotNet.ProjectModel/WorkspaceContext.cs

```diff
@@ -445,7 +445,13 @@ private static string NormalizeProjectPath(string path)
             if (File.Exists(path) &&
                 string.Equals(Path.GetFileName(path), Project.FileName, StringComparison.OrdinalIgnoreCase))
             {
-                return Path.GetFullPath(Path.GetDirectoryName(path));
+                string directoryName = Path.GetDirectoryName(path);
```

Comment:
I do think that would be appropriate. 

These functional tests have been pretty valuable, and this is a callsite level issue which could be regressed without failing this unit tests.